dali/operators/segmentation/select_masks.cc

```diff
@@ -0,0 +1,226 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/operators/segmentation/select_masks.h"
+
+#include <utility>
+#include "dali/kernels/common/type_erasure.h"
+#include "dali/core/static_switch.h"
+
+namespace dali {
+
+DALI_SCHEMA(segmentation__SelectMasks)
+    .DocStr(R"(Selects a subset of mask by their mask ids.
+
+The operator expects three inputs describing multiple segmentation mask polygons and a list of selected mask ids.
+
+Each sample can contain several masks, and each mask can be composed by several polygons. The masks are described 
+by the inputs ``mask_meta`` and ``mask_coords``. The format of this data is the same as the one produced by COCOReader.
+
+The operator receives a list of mask ids that are to be selected for each sample, and produces
+output ``mask_meta`` and ``mask_coords`` where only the masks corresponding to selected mask ids are present.
+
+**Examples:**
+
+Let us assume the following input mask, where symbolic coordinates are used for a clearer example::
+
+    ``mask_coords`` = [[x0, y0], [x1, y1], [x2, y2], [x3, y3], [x4, y4], [x5, y5], [x6, y6], [x7, y7], [x8, y8], [x9, y9]]
+    ``mask_meta`` = [[0, 0, 2], [1, 3, 6], [2, 7, 9]
+
+Example 1: Selecting a single mask with id ``1``, maintaining the original id::
+
+    ``mask_ids`` = [1], ``reindex_masks`` = False
+    ``out_mask_meta`` = [[1, 0, 4]]
+    ``out_mask_coords`` = [[x3, y3], [x4, y4], [x5, y5], [x6, y6]]
+
+Example 2: Selecting two out of the three mask, reindexing the mask ids to follow the order of the input mask ids::
+
+    ``mask_ids`` = [2, 0]
+    ``reindex_masks`` = True
+    ``out_mask_meta`` = [[0, 3, 5], [1, 0, 2]]
+    ``out_mask_coords`` = [[x0, y0], [x1, y1], [x2, y2], [x7, y7], [x8, y8], [x9, y9]]
+)")
+    .NumInput(3)
+    .NumOutput(2)
+    .InputDox(0, "mask_ids", "1D TensorList of int",
+              R"code(List of mask identifiers that are to be selected.)code")
+    .InputDox(
+        1, "mask_meta", "2D TensorList of int",
+        R"code(Each row describes a single mask with three values, ``mask_idx, start_idx, end_idx``,
+where ``start_idx`` and ``end_idx`` refer to indices in ``mask_coords``.)code")
+    .InputDox(2, "mask_coords", "2D TensorList of float or int",
+              R"code(Each row contains a single ``x, y`` coordinate.)code")
```

Comment:
```suggestion
              R"code(Each row contains a single set of coordinates, kike ``x, y`` or ``x, y, z``.code")
```
...and, of course, we should support more than 2D points.